### PR TITLE
[manta]Add strip profile to production

### LINKED ISSUE #325: Remove strip operation in CI
We can remove strip command in CI due to this pr stablized: https://github.com/rust-lang/cargo/pull/10088
But not sure rust 1.58 will include this feature.

### PR DESCRIPTION
**Pull request by Dengjianping**
Signed-off-by: Dengjianping <djptux@gmail.com>

## Description

Remove `strip` from CI, and add `strip` profile to production.

https://doc.rust-lang.org/stable/cargo/reference/profiles.html#strip

closes: #325

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests.
- [ ] Updated relevant documentation in the code.
- [ ] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. If this number is updated, then the spec_version must also be updated 
- [ ] Verify benchmarks & weights have been updated for any modified runtime logics
- [ ] If import a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [ ] If needed, update our Javascript/Typescript APIs. These APIs are offcially used by exchanges or community developers.
- [ ] If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inhreited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by Garandor** [comment]
Seems like srtool doesn't want the runtime to be built without strip:
https://github.com/Manta-Network/Manta/runs/5572932228?check_suite_focus=true#step:4:340

**Comment by Dengjianping** [comment]
> Seems like srtool doesn't want the runtime to be built without strip: https://github.com/Manta-Network/Manta/runs/5572932228?check_suite_focus=true#step:4:340

Yeah, srtool is blocking us, I just created an issue in srtool, not sure when it will be fixed.
https://github.com/paritytech/srtool/issues/40

**Comment by Garandor** [comment]
one of us might be misinterpreting the error
Caused by:
  ║ feature `strip` is required

The way I understand it It's not about not being stabilized (the last part of the error message is misleading here), it's about the CI build in your change trying to make a binary without `strip` enabled.
It'd be logical to me if srtool is preventing wasm runtime unstablenessbuild without `strip` as those should never end up onchain.

So it'd be about srtool allowing unstripped builds, not about unstableness

EDIT: nvm, just saw that `strip` was being invoked as a standalone command here https://github.com/Manta-Network/Manta/pull/449/files#diff-84486036e95f7e22168734603cbdd311736c85fb23a0d0e0660988a67d3605ecL163
so the misunderstanding was on my part

**Comment by stechu** [comment]
@Garandor  can you give an explicit review of this PR (approve or request change)?

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ members = [
 inherits = "release"
 codegen-units = 1
 lto = true
+strip = "symbols"
```
Comment (Garandor) [inline Cargo.toml:17]: Do we use `--production` in the CI somewhere or is this manual exeuction only?
If so, is the release checklist adapted so we don't mistakenly publish a `--release` build

Comment (Dengjianping) [reply Cargo.toml:17]: Our CI has been switched to `production` profile.

**Comment by Dengjianping** [comment]
> 

We have to wait, it seems parity will publish a release(0.9.20) for fixing this problem.

**Review by Garandor — APPROVED** [review]
LGTM for 3.2.1

**Review by Garandor — CHANGES_REQUESTED** [review]
merge conflicts

**Review by Garandor — APPROVED** [review]
LGTM

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**